src/v/storage/disk_log_impl.cc

```diff
@@ -1701,15 +1701,10 @@ log make_disk_backed_log(
 
 ss::future<reclaim_size_limits>
 disk_log_impl::estimate_reclaim_size(compaction_config cfg) {
```

Comment:
> However, even when retention is disabled we may still want to delete data that has been uploaded to cloud storage.

Just making sure I understand the behavior before this: did we previously not locally GC for compacted topics? Or did we just not estimate the size, and therefore not prioritize running GC code as highly as we should've been

Reply:
Before there was no estimation. Now we can estimate, but we aren't yet using that in the scheduling decision. Also, generally, this is work is tackling _retention_ policies, so it is orthogonal to compaction. We aren't estimating for compaction.

Reply:
Got it, thanks for clarifying